### PR TITLE
feat: add nicer constants

### PR DESCRIPTION
**Pull request by marktoda**

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,9 +1,19 @@
...
+export const PERMIT2_ADDRESS = '0x000000000022D473030F116dDEE9F6B43aC78BA3'
+
 export const MaxUint160 = BigNumber.from('0xffffffffffffffffffffffffffffffffffffffff')
 export const MaxUint256: BigNumber = BigNumber.from(
   '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
 )
 export const MaxUint48: BigNumber = BigNumber.from('0xffffffffffff')
 
+// alias max types for their usages
+export const MaxAllowanceTransferAmount = MaxUint160
```
Comment (zzmp) [inline src/constants.ts:12]: You have inconsistent types in this file. IMO they should all be inferred, and there should be no `: BigNumber` explicit typing.

File: src/constants.ts [hunk 2]
```diff
@@ -1,9 +1,19 @@
 import { BigNumber } from '@ethersproject/bignumber'
 
+export const PERMIT2_ADDRESS = '0x000000000022D473030F116dDEE9F6B43aC78BA3'
+
 export const MaxUint160 = BigNumber.from('0xffffffffffffffffffffffffffffffffffffffff')
 export const MaxUint256: BigNumber = BigNumber.from(
   '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
 )
 export const MaxUint48: BigNumber = BigNumber.from('0xffffffffffff')
```
Comment (zzmp) [inline src/constants.ts:9]: nit: order these by size: 48, 160, 256.

File: src/constants.ts (hunk 1 above)
Comment (zzmp) [inline src/constants.ts:12]: nit: either order alphabetically or order by usage

File: src/constants.ts [hunk 3]
```diff
@@ -1,9 +1,19 @@
 import { BigNumber } from '@ethersproject/bignumber'
 
+export const PERMIT2_ADDRESS = '0x000000000022D473030F116dDEE9F6B43aC78BA3'
+
 export const MaxUint160 = BigNumber.from('0xffffffffffffffffffffffffffffffffffffffff')
```
Comment (zzmp) [inline src/constants.ts:5]: imo you should no longer export these unsemantic values (although I think @tinaszheng disagrees)

File: src/signatureTransfer.ts [hunk 4]
```diff
@@ -111,8 +113,12 @@ export abstract class SignatureTransfer {
     chainId: number,
     witness?: Witness
   ): PermitTransferFromData | PermitBatchTransferFromData {
+    invariant(permit.deadline <= MaxSigDeadline, 'INVALID_SIG_DEADLINE')
```
Comment (zzmp) [inline src/signatureTransfer.ts:116]: I did not think operator overloading was implementable in TS. Don't you need to use `MaxSigDeadline.gte(permit.deadline)`?

Same with other invariant usages.

Comment (zzmp) [inline src/signatureTransfer.ts:116]: Better than `INVALID`, I'd like `SIG_DEADLINE_OUT_OF_RANGE`.
Best would be to throw a `RangeError`, but I don't think that `tiny-invariant` supports errors, only strings.

Same with other invariant usages.

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — CHANGES_REQUESTED** [review]
add tests for out-of-range values that will pass to ensure the checks pass.
this will check if the operator overloading is valid or not.

File: src/constants.ts (hunk 1 above)
Comment (marktoda) [reply src/constants.ts:12]: currently theyre grouped by which flow theyre referencing: first are AllowanceTransfer flow, second are SignatureTransfer flow

Comment (marktoda) [reply src/constants.ts:12]: ill add comment / space to clarify

File: src/constants.ts (hunk 3 above)
Comment (marktoda) [reply src/constants.ts:5]: yea i think these are still useful

File: src/constants.ts [hunk 5]
```diff
@@ -1,9 +1,20 @@
...
-export const MaxUint256: BigNumber = BigNumber.from(
-  '0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff'
-)
-export const MaxUint48: BigNumber = BigNumber.from('0xffffffffffff')
+export const MaxUint256 = BigNumber.from('0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff')
+
+// alias max types for their usages
+// allowance transfer types
+export const MaxAllowanceTransferAmount = MaxUint160
+export const MaxSignatureTransferAmount = MaxUint256
```
Comment (zzmp) [inline src/constants.ts:12]: this is a signature transfer type

File: src/constants.ts [hunk 6]
```diff
@@ -1,9 +1,20 @@
...
+export const MaxAllowanceExpiration = MaxUint48
+
+// signature transfer types
+export const MaxOrderedNonce = MaxUint48
```
Comment (zzmp) [inline src/constants.ts:16]: this is an allowance transfer type

**Review by zzmp — APPROVED** [review]
I think some of the aliased types are grouped incorrectly.
otherwise lgtm

**Pull request merged**